pandas/io/json/_json.py

```diff
@@ -972,10 +972,8 @@ def _try_convert_to_date(self, data):
         for date_unit in date_units:
             try:
                 new_data = to_datetime(new_data, errors="raise", unit=date_unit)
-            except ValueError:
+            except (ValueError, OverflowError):
```

Comment:
Actually should we be catching an `OutOfBoundsDatetime` exception here? Not sure if test coverage is hitting that but I think easy enough to trigger (just try a date like "9999-01-01")

Out of curiosity what causes the OverflowError?

Reply:
I think it is cast_to_unit being called inside array_to_datetime, but would have to check.

`OutOfBoundsDatetime` subclasses `ValueError` so that is already caught.